docs/odh/oci-model-storage.md

```diff
@@ -8,15 +8,228 @@ which also explains how to deploy models from OCI images.
+> Feel free to try it with any other model that can work with the vLLM server.
+
+Notice that model files are copied into `/models` inside the final container. KServe
+expects this path to exist in the OCI image and also expects the model files to
+be inside it.
+
+Also, notice that `ubi9-micro` is used as a base container of the final image.
+Empty images, like `scratch` cannot be used, because KServe needs to configure the model image
+with a command to keep it alive and ensure the model files remain available in
+the pod. Thus, it is required to use a base image that provides a shell.
+
+Finally, notice that ownership of the copied model files is changed to the `root`
+group, and also read permissions are granted. This is important, because OpenShift
+runs containers with a random user ID and with the `root` group ID. The adjustment
+of the group and the privileges on the model files ensures that the model server
+can access them.
+
+Create the OCI container image of the Granite model using Podman, and upload it to
+a registry. For example, using Quay as the registry:
+```shell
+podman build --format=oci --squash \
+  --build-arg repo_id=ibm-granite/granite-3.0-2b-instruct \
+  -t quay.io/<user_name>/<repository_name>:<tag_name> .
+
+podman push quay.io/<user_name>/<repository_name>:<tag_name>
+```
+
+It is important to use the `--squash` flag to prevent the final image having
+the double size of the model.
+
+> [!TIP]
+> When uploading your container image, if your repository is private, ensure you
+> are authenticated to the registry.
+
+### Deploying the Granite model using the generated OCI image
+
+Start by creating a namespace to deploy the model:
+```shell
+oc new-project oci-model-example
+```
+
+In the newly created namespace, you need to create a `ServingRuntime` resource
+configuring vLLM model server. The ODH project provides templates with
+configurations for some model servers, which you can list with the following
+command:
+```shell
+oc get templates -n opendatahub
+
+NAME                                 DESCRIPTION                                                                        PARAMETERS    OBJECTS
+caikit-standalone-serving-template   Caikit is an AI toolkit that enables users to manage models through a set of...    0 (all set)   1
+caikit-tgis-serving-template         Caikit is an AI toolkit that enables users to manage models through a set of...    0 (all set)   1
+kserve-ovms                          OpenVino Model Serving Definition                                                  0 (all set)   1
+ovms                                 OpenVino Model Serving Definition                                                  0 (all set)   1
+tgis-grpc-serving-template           Text Generation Inference Server (TGIS) is a high performance inference engin...   0 (all set)   1
+vllm-runtime-template                vLLM is a high-throughput and memory-efficient inference and serving engine f...   0 (all set)   1
```

Comment:
This list is changing frequently, like adding a new accelerator image or similar, I would just keep one single vLLM entry here (for example currently we have [4 vLLM templates](https://github.com/opendatahub-io/odh-model-controller/tree/incubating/config/runtimes))

```suggestion
vllm-runtime-template                vLLM is a high-throughput and memory-efficient inference and serving engine f...   0 (all set)   1
```

Btw in the official documentation I would try to use Dashboard only: with new ODH Connection it should be possible to configure a OCI connection too (I still need to try)

Reply:
just adding might be enough as well, this way we hide other templates that are not relevant to this example
```sh
$ command
...
vllm-runtime-template                vLLM is a high-throughput and memory-efficient inference and serving engine f...   0 (all set)   1
...
```

Reply:
I changed the text so that the output is only for the runtime used.